js/interface/add-markers.js

```diff
@@ -0,0 +1,500 @@
+
+      if (fromLoad) {
+        // Manually removes markers
+        this.removeMarkers()
+
+        this.mappedMarkerData.forEach((marker, index) => {
+          if (marker.data.floor === this.selectedMarkerData.floor.name) {
+            const markersLength = this.flPanZoomInstance.markers.getAll().length
+            markerElem = $("<div id='" + marker.id + "' class='marker' data-name='" + marker.data.name + "' style='left: -15px; top: -15px; position: absolute;'><i class='" + marker.data.icon + "' style='color: " + marker.data.color + "; font-size: " + marker.data.size + ";'></i><div class='active-state' style='background-color: " + marker.data.color + ";'></div></div>")
+            this.markerElemHandler = new Hammer(markerElem.get(0))
+            this.flPanZoomInstance.markers.set([Fliplet.UI.PanZoom.Markers.create(markerElem, { x: marker.data.positionx, y: marker.data.positiony, name: marker.data.name, id: marker.id })])
+            this.markerElemHandler.on('tap', this.onMarkerHandler)
+          }
+        })
+        return
+      }
+
+      if (options.existingMarker) {
+        options.existingMarker.update({x: options.x, y: options.y}, true)
+        this.updateMarkerCoordinates({
+          x: options.x,
+          y: options.y,
+          marker: options.existingMarker.vars
+        })
+        $('#' + options.id).addClass('active')
+      } else {
+        const markersLength = this.tappedMarkerId || this.flPanZoomInstance.markers.getAll().length
+        markerElem = $("<div id='" + this.selectedMarkerData.marker.id + "' class='marker' data-name='" + this.selectedMarkerData.marker.data.name + "' style='left: -15px; top: -15px; position: absolute;'><i class='" + this.selectedMarkerData.marker.data.icon + "' style='color: " + this.selectedMarkerData.marker.data.color + "; font-size: " + this.selectedMarkerData.marker.data.size + ";'></i><div class='active-state' style='background-color: " + this.selectedMarkerData.marker.data.color + ";'></div></div>")
+        this.markerElemHandler = new Hammer(markerElem.get(0))
+        this.flPanZoomInstance.markers.set([Fliplet.UI.PanZoom.Markers.create(markerElem, { x: options.x, y: options.y, name: this.selectedMarkerData.marker.data.name, id: this.selectedMarkerData.marker.id })])
+        $('#marker-' + markersLength).addClass('active')
+        this.tappedMarkerId = undefined
+      }
+      
+      this.markerElemHandler.on('tap', this.onMarkerHandler)
+    },
+    updateMarkerCoordinates(coordinates) {
+      if (!coordinates) {
+        return
+      }
+
+      this.mappedMarkerData.forEach((marker, index) => {
+        if (marker.data.name === coordinates.marker.name) {
+          this.mappedMarkerData[index].data.positionx = coordinates.x
+          this.mappedMarkerData[index].data.positiony = coordinates.y
+        }
+      })
+      this.saveDebounced()
+    },
+    attachEventHandler() {
+      this.pzHandler.on('tap', this.onTapHandler)
+    },
+    detachEventHandlers() {
```

Comment:
`attachEventHandler` doesn't have `s` whereas `detachEventHandlers` has `s`. Singular/plural should be consistent. I'd use plural in both cases. There's no benefit to keeping the singular/plural technically correct, but there is benefit to keeping them consistent.